interfaces/cython/pyproject.toml

```diff
@@ -1,3 +1,21 @@
 [build-system]
 requires = ["setuptools>=43.0.0", "wheel", "Cython>=0.29.12"]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "cantera"
+dynamic = [
+    "authors",
+    "classifiers",
+    "description",
+    "keywords",
+    "license",
+    "optional-dependencies",
+    "readme",
+    "scripts",
+    "urls",
+    "version",
+]
+dependencies = [
+    "numpy>=1.19.3",
+]
```

Comment:
I think this should be coming from `setup.cfg`, is it not?

Reply:
Just trying things. I'm not finding the supposed ABI compatibility of the compiled module with older NumPy versions.

Reply:
I'll experiment with this tomorrow too

Reply:
Well, from the latest runs, it seems like backwards compatibility is working as expected when building with NumPy 2.0.0. The problem is with the builds for Python 3.8 using NumPy 1.24 and then running with NumPy 1.19. And now I've finally read [this page](https://numpy.org/devdocs/dev/depending_on_numpy.html#adding-a-dependency-on-numpy) enough times to understand that this is expected not to work and we have to go the other way around in this case.

Reply:
Ok, got it. The dependency, at least in the sdist at wheel build time, is that "oldest-supported-numpy" option they mention. But good to know that it needs to depend on the Python versions that NumPy supports

Reply:
Yep. It sounds like this _should_ be a lot easier going forward -- a build-time dependency on NumPy 2.x will provide backwards compatibility with NumPy 1.x and also forward compatibility with future 2.x releases. If we want to get ahead of things, I suppose we could set the version requirement to `numpy<3`.